app/scripts/modules/core/src/cloudProvider/versionSelection/versionSelector.component.ts

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+import { IModalInstanceService } from 'angular-ui-bootstrap';
+
+import { ACCOUNT_SERVICE } from 'core/account/account.service';
+import { CLOUD_PROVIDER_REGISTRY } from 'core/cloudProvider/cloudProvider.registry';
+
+import { IController, IScope, module } from 'angular';
+
+export class VersionSelectorCtrl implements IController {
+  constructor(private $scope: IScope,
```

Comment:
Nothing really wrong with `$scope`, but since starting to use TS classes we generally reference properties of the controller class instead. In your controller's template, you can just swap `this` for `ctrl` (because of `versionSelectorCtrl as ctrl`). I.e., on your template, you can just reference `ctrl.versionOptions`.